src/core/components/app.jsx

```diff
@@ -10,6 +11,13 @@ export default class App extends React.Component {
     return Component ? Component : ()=> <h1> No layout defined for &quot;{layoutName}&quot; </h1>
   }
 
+  componentDidMount() {
+    const sys = this.props
+    window.onhashchange = function() {
```

Comment:
This listener should only be registered if `deepLinking` is enabled, for performance reasons.